Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,233 @@
-﻿namespace BootCamp.Chapter
+﻿using Microsoft.VisualBasic.CompilerServices;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+        static string defaultReturn = "N/A.";
+
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances != null && peopleAndBalances.Length != 0)
+            {
+                var highNames = new List<string>();
+                int highBal = 0;
+
+                foreach (var str in peopleAndBalances)
+                {
+                    var arr = str.Split(", ");
+                    var name = arr[0];
+                    int balance = 0;
+                    for (int i = 1; i < arr.Length; i++)
+                    {
+                        var ok = int.TryParse(arr[i], out int number);
+                        if (ok)
+                            balance = Math.Max(balance, number);
+                    }
+                    if (balance > highBal)
+                    {
+                        highNames.Clear();
+                        highNames.Add(name);
+                        highBal = balance;
+                    }
+                    else if (balance == highBal)
+                    {
+                        highNames.Add(name);
+                    }
+                }
+
+                StringBuilder sb = new StringBuilder();
+                var highBalStr = FormatValue(highBal);
+
+                BuildNamesString(sb, highNames);
+                sb.Append($" had the most money ever. {highBalStr}.");
+
+                return sb.ToString();
+            }
+            return defaultReturn;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances != null && peopleAndBalances.Length != 0)
```

Comment:
Invert if and return default here- will reduce nesting.

Reply:
Inverted all if statements